udata/api/__init__.py

```diff
@@ -149,6 +173,38 @@ def set_api_language():
         g.lang_code = request.args['lang']
 
 
+def extract_name_from_path(path):
+    """Return a readable name from a URL path.
+
+    Useful to log requests on Piwik with categories tree structure.
+    See: http://piwik.org/faq/how-to/#faq_62
+    """
+    base_path, query_string = path.split('?')
+    infos = base_path.strip('/').split('/')[2:]  # Removes api/version.
+    if len(infos) > 1:  # This is an object.
+        name = '{category} / {name}'.format(
+            category=infos[0].title(),
+            name=infos[1].replace('-', ' ').title()
+        )
+    else:  # This is a collection.
+        name = '{category}'.format(category=infos[0].title())
+    return name
+
+
+@apiv1.after_request
+def collect_stats(response):
+    action_name = extract_name_from_path(request.full_path)
+    params = {
+        'action_name': urllib.quote(action_name),
+        'user_ip': request.remote_addr
+    }
+    if current_user.is_authenticated():
+        params['_id'] = str(hash(current_user.email)).strip('-')
```

Comment:
Maybe the solution is to use ``uid`` instead of ``_id``: http://piwik.org/docs/user-id/

But this is only for logged in users (which is fine beacause we only track logged actions serverside)

Piwik can set a cookie ID: http://piwik.org/faq/how-to/faq_175/

I can see a ``PIWIK_SESSID`` in cookies but I don't know if it's reusable.